src/Avalonia.X11/NativeDialogs/GtkNativeFileDialogs.cs

```diff
@@ -102,23 +102,23 @@ void Dispose()
             return tcs.Task;
         }
         
-        public async Task<string[]> ShowFileDialogAsync(FileDialog dialog, IWindowImpl parent)
+        public async Task<string[]> ShowFileDialogAsync(FileDialog dialog, Window parent)
         {
             await EnsureInitialized();
             return await await RunOnGlibThread(
-                () => ShowDialog(dialog.Title, parent,
+                () => ShowDialog(dialog.Title, parent?.PlatformImpl,
```

Comment:
I think we should save the platform impl while still being on the UI thread.

Reply:
Good point, I really wonder if we could have actual platform tests written at some point.